xtra-libp2p/src/endpoint.rs

```diff
@@ -264,25 +295,46 @@ impl Endpoint {
         peer: PeerId,
         protocols: Vec<&'static str>,
     ) -> Result<(&'static str, Substream), Error> {
-        let (control, _) = self
-            .controls
-            .get_mut(&peer)
-            .ok_or(Error::NoConnection(peer))?;
-
-        let stream = control.open_stream().await?;
-
-        let (protocol, stream) = tokio::time::timeout(
-            self.connection_timeout,
-            multistream_select::dialer_select_proto(stream, protocols, Version::V1),
-        )
-        .await
-        .map_err(|_timeout| Error::NegotiationTimeoutReached)?
-        .map_err(Error::NegotiationFailed)?;
-
-        Ok((
-            protocol,
-            Substream::new(stream, protocol, libp2p_core::Endpoint::Dialer),
-        ))
+        let substream = {
```

Comment:
nit - probably would be easier to follow if we named this `result` or `substream_result`